File: JS/check_braces.js
const fs = require('fs');
const content = fs.readFileSync('style.css', 'utf8');

const stack = [];
const lines = content.split(/\r?\n/);
let extra = false;
for (let i = 0; i < lines.length; i++) {
  const line = lines[i];
  const lineNum = i + 1;
  for (let j = 0; j < line.length; j++) {
    const char = line[j];
    if (char === '{') {
      stack.push({ lineNum, line });
    } else if (char === '}') {
      if (stack.length === 0) {
        console.log(`Extra closing brace at line ${lineNum}: ${line}`);
        extra = true;
      } else {
        stack.pop();
      }
    }
  }
}

if (stack.length > 0) {
  console.log("Unclosed braces:");
  stack.forEach(item => {
    console.log(`  Line ${item.lineNum}: ${item.line}`);
  });
} else if (!extra) {
  console.log("All braces match perfectly!");
}

File: JS/check_braces.py
with open(r"c:\Users\pauli\OneDrive\Documents\Desktop\nemesis3\style.css", "r", encoding="utf-8") as f:
    content = f.read()

# Check curly braces
stack = []
lines = content.splitlines()
for line_num, line in enumerate(lines, 1):
    for char_idx, char in enumerate(line):
        if char == '{':
            stack.append((line_num, char_idx, line))
        elif char == '}':
            if not stack:
                print(f"Extra closing brace at line {line_num}, col {char_idx}: {line}")
            else:
                stack.pop()

if stack:
    print(f"Unclosed braces:")
    for item in stack:
        print(f"  Line {item[0]}: {item[2]}")
else:
    print("All braces match perfectly!")
